src/libraries/System.Text.Json/Common/JsonSourceGenerationOptionsAttribute.cs

```diff
@@ -15,11 +15,58 @@ namespace System.Text.Json.Serialization
 #endif
     sealed class JsonSourceGenerationOptionsAttribute : JsonAttribute
     {
+        /// <summary>
+        /// Constructs a new <see cref="JsonSourceGenerationOptionsAttribute"/> instance.
+        /// </summary>
+        public JsonSourceGenerationOptionsAttribute() { }
+
+        /// <summary>
+        /// Constructs a new <see cref="JsonSourceGenerationOptionsAttribute"/> instance with a predefined set of options determined by the specified <see cref="JsonSerializerDefaults"/>.
+        /// </summary>
+        /// <param name="defaults">The <see cref="JsonSerializerDefaults"/> to reason about.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Invalid <paramref name="defaults"/> parameter.</exception>
+        public JsonSourceGenerationOptionsAttribute(JsonSerializerDefaults defaults)
+        {
+            // Constructor kept in sync with equivalent overload in JsonSerializerOptions
+
+            if (defaults is JsonSerializerDefaults.Web)
+            {
+                PropertyNameCaseInsensitive = true;
+                PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase;
+                NumberHandling = JsonNumberHandling.AllowReadingFromString;
+            }
+            else if (defaults is not JsonSerializerDefaults.General)
+            {
+                throw new ArgumentOutOfRangeException(nameof(defaults));
+            }
+        }
+
+        /// <summary>
+        /// Defines whether an extra comma at the end of a list of JSON values in an object or array
+        /// is allowed (and ignored) within the JSON payload being deserialized.
+        /// </summary>
+        public bool AllowTrailingCommas { get; set; }
+
+        /// <summary>
+        /// Specifies a list of custom converter types to be used.
+        /// </summary>
+        public Type[]? Converters { get; set; }
+
+        /// <summary>
+        /// The default buffer size in bytes used when creating temporary buffers.
+        /// </summary>
+        public int DefaultBufferSize { get; set; }
```

Comment:
Because the attribute is only being read by Roslyn, the value of the property is only being consulted if explicitly set by the user in the attribute declaration. I briefly considered making this and other properties nullable, however that would be a departure from convention already used in properties shipped in this attribute, so went with consistency instead. The other alternative would be to explicitly set the properties to equal the `JsonSerializerOptions` default, however that would necessitate a bit of duplication or refactoring without providing any clear benefit.

Reply:
Then shouldn't the XML comment call out what a value of 0 means, as is done below for MaxDepth?

Reply:
I would probably opt for removing that reference in `MaxDepth`, making all of them deliberately ambiguous and just have them point to the corresponding `JsonSerializerOptions` properties.

Reply:
We need to call out somewhere in the docs what it means to not provide a value for these.

Reply:
I'll follow up with a PR once we conclude P7 work.